Makefile

```diff
@@ -412,8 +412,8 @@ ifeq ($(USE_DIST_KVSTORE), 1)
 endif
 
 #sparse-matrix
+SPARSE_MATRIX_DIR =  $(ROOTDIR)/3rdparty/sparse-matrix
 ifeq ($(USE_BLAS), mkl)
-	SPARSE_MATRIX_DIR =  $(ROOTDIR)/3rdparty/sparse-matrix
```

Comment:
I thought we only need sparse-matrix when on MKL, is that not correct?

Reply:
yes, you are right.
I just move it back under MKL macro.